crates/sui-data-ingestion-core/src/workers.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use async_trait::async_trait;
+use sui_types::full_checkpoint_content::CheckpointData;
+
+#[async_trait]
+pub trait Worker: Send + Sync {
+    async fn process_checkpoint(&self, checkpoint: CheckpointData) -> Result<()>;
+    async fn save_progress(&self) -> bool {
+        true
+    }
+}
```

Comment:
Could we get some documentation around the expectations implementors should have? As in:
- What are each of these methods for?
- What ordering guarantees exist
- Can checkpoints be given simultaneously so that things can operate in parallel?

etc